osc.js

```diff
@@ -1,87 +1,574 @@
-var instance_skel = require('../../instance_skel')
-var GetUpgradeScripts = require('./upgrades')
-var OSC = require('osc')
+const instance_skel = require('../../instance_skel')
+const GetUpgradeScripts = require('./upgrades')
+const OSC = require('osc')
 
-function instance(system, id, config) {
-	var self = this
+class instance extends instance_skel {
+	constructor(system, id, config) {
+		// super-constructor
+		super(system, id, config)
+		this.actions() // export actions
+		return this
+	}
 
-	// super-constructor
-	instance_skel.apply(this, arguments)
+	GetUpgradeScripts = GetUpgradeScripts
 
-	self.actions() // export actions
+	updateConfig(config) {
+		this.config = config
```

Comment:
I typically try to compare both the old and new hosts and ports to see if `this.osc_server_init()` needs to be run on account of a change to those config fields.